### DISCUSSION TITLE
Using a matrix defined input for a custom action

### DISCUSSION THREAD

**Discussion by radu-matei** [community/community#25899] [Actions]
<p>I want to use a custom action and provide a different value to an action input parameter based on the matrix element - for example, a platform specific URL.</p>
<p>Is there any way I can do that with the current matrix syntax?</p>
<p>For example, the following will use the Windows-only parameter for a Linux build and vice-versa:</p>
<pre><code>runs-on: ${{ matrix.os }}
strategy:
  matrix:
    os: [ubuntu-latest, windows-latest]
    custom: [linux-only, windows-only]
steps:
  - uses: my-custom-action
    with:
      myInput: ${{ matrix.custom }}
</code></pre>
<p>I know I can use the exclude rules for the matrix, but it really does seem cumbersome for more complex scenarios.</p>
<p>Additionally, there might be multiple parameters that have platform specific values I would like to set, which means the exclusion solution will not work here.</p>
<p>Any pointers to how this could be done easier?</p>
<p>Thanks!</p>

**Comment by ethomson** [comment]
<p>It doesn’t seem like you’d want a matrix here.  I think I would approach this by having a script that <a href="https://help.github.com/en/articles/development-tools-for-github-actions#logging-commands" rel="nofollow noopener">sets the appropriate values</a> based on the operating system in the matrix, then uses those output values as inputs into the other actions.  For example:</p>
<pre><code>runs-on: ${{ matrix.os }}
strategy:
  matrix:
    os: [ubuntu-latest, windows-latest]
steps:
- run: |
    if ["${{ matrix.os }}" = "ubuntu-latest"]; then
      echo ::set-output name=url::http://linux_url/
      echo ::set-output name=other::foo
    fi
    if ["${{ matrix.os }}" = "windows-latest"]; then
      echo ::set-output name=url::http://windows_url/
      echo ::set-output name=other::bar
    fi
  shell: bash
- uses: custom/action_one@master
  with:
    url: ${{ steps.configure.outputs.url }}
- uses: custom/action_two@master
  with:
    other: ${{ steps.configure.outputs.other }}
</code></pre>

**Comment by wipe2238** [comment]
<p>Shouldn’t <code>run</code> step have <code>id: configure</code> set?</p>

**Comment by chrispat** [comment, marked as answer]
<p>You sould also be able to do something like this</p>
<pre><code>runs-on: ${{ matrix.config.os }}
strategy:
  matrix:
    config:
    - {os: ubuntu-latest, url: "http://linuxurl" }
    - {os: windows-latest, url: "http://windowsurl" }
steps:
  - uses: my-custom-action
    with:
      myInput: ${{ matrix.config.url }}
</code></pre>

**Comment by radu-matei** [comment]
<p>This is exactly what I was looking for, thanks a lot!</p>
